File: java/src/main/java/deltix/luminary/implementation/DecoratorPropertyNode.java
package deltix.luminary.implementation;

import deltix.luminary.Literal;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class DecoratorPropertyNode extends PropertyNode {
    private Literal defaultValue = null;

    public DecoratorPropertyNode(String name, String type, List<String> comments) {
        super(name, type, comments);
    }

    @Nullable
    public Literal getDefault() {
        return defaultValue;
    }

    void setDefault(@NotNull Literal literal) {
        defaultValue = literal;
    }
}

File: csharp/main/Implementation/DecoratorPropertyNode.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary.Implementation
{
	internal class DecoratorPropertyNode : PropertyNode
	{
		public DecoratorPropertyNode([NotNull] String name, [NotNull] String type, [CanBeNull] List<String> comments)
			: base(name, type, comments)
		{
		}

		[CanBeNull]
		public Literal Default { get; internal set; }
	}
}
